sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessor.java

```diff
@@ -64,28 +50,22 @@
      * @param eventHubAsyncClient The {@link EventHubAsyncClient}.
      * @param consumerGroupName The consumer group name used in this event processor to consumer events.
      * @param partitionProcessorFactory The factory to create new partition processor(s).
-     * @param initialEventPosition Initial event position to start consuming events.
-     * @param partitionManager The partition manager.
-     * @param eventHubName The Event Hub name.
+     * @param initialEventPosition The event position to start processing events from a partition if no checkpoint is
+     * available for the partition.
+     * @param partitionManager The partition manager this Event Processor will use for reading and writing partition
+     * ownership and checkpoint information.
      */
     EventProcessor(EventHubAsyncClient eventHubAsyncClient, String consumerGroupName,
         PartitionProcessorFactory partitionProcessorFactory, EventPosition initialEventPosition,
-        PartitionManager partitionManager,
-        String eventHubName) {
-        this.eventHubAsyncClient = Objects
-            .requireNonNull(eventHubAsyncClient, "eventHubAsyncClient cannot be null");
-        this.consumerGroupName = Objects
-            .requireNonNull(consumerGroupName, "consumerGroupname cannot be null");
-        this.partitionProcessorFactory = Objects
-            .requireNonNull(partitionProcessorFactory, "partitionProcessorFactory cannot be null");
-        this.partitionManager = Objects
-            .requireNonNull(partitionManager, "partitionManager cannot be null");
-        this.initialEventPosition = Objects
-            .requireNonNull(initialEventPosition, "initialEventPosition cannot be null");
-        this.eventHubName = Objects
-            .requireNonNull(eventHubName, "eventHubName cannot be null");
+        PartitionManager partitionManager) {
+
         this.identifier = UUID.randomUUID().toString();
         logger.info("The instance ID for this event processors is {}", this.identifier);
+        this.partitionPumpManager = new PartitionPumpManager(partitionManager, partitionProcessorFactory,
```

Comment:
![](https://media1.giphy.com/media/gdM6D5CAc1puw/giphy.gif)

Reply:
😄 I used the same name Track 1 was using which I am not entirely satisfied with. Do you have a better name suggestion?

Reply:
Unfortunately, not a wonderful one.   Maybe something like `PartitionReaderManager` or similar?    Either way, I think you're good.  I just have a fun mental image every time I read about their "pumps"

Reply:
They definitely have a lot of "pumps".. (ie. PartitionPump). 🤣

Reply:
MessageDispatcher?

Reply:
I'd be a bit cautious with the word `Message` in the Event Hubs context.  Easy to confuse whether it acts on events or AMQP messages.... but the `Dispatcher` part of that, I really like!

Reply:
... and because I feel that it is a moral imperative, each time I visit this PR:

![image](https://user-images.githubusercontent.com/913445/63434618-6f1ab000-c3f3-11e9-9de4-b16ef1d19005.png)